### PR TITLE
fix: Fix latching MOVEMENT-1429

### PR DESCRIPTION
**Pull request by ccpowers**

### CODE REVIEW

File: dwa_local_planner/src/dwa_planner_ros.cpp [hunk 1]
```diff
@@ -344,8 +360,11 @@ namespace dwa_local_planner {
   }
 
   bool DWAPlannerROS::cancel() {
-  canceled_ = true;
-  return true;
+    // if we cancel, reset the latching
+    ROS_DEBUG_NAMED("dwa_local_planner", "Clearing latching in dwa cancel. How did we get here?");
```
Comment (27Apoorva) [inline dwa_local_planner/src/dwa_planner_ros.cpp]: are we getting here??

Comment (ccpowers) [reply dwa_local_planner/src/dwa_planner_ros.cpp]: no, this is wrapped by the wrapper_local_planner in move base flex, and never called. the log line is just to warn us if that changes in the future. i'll change it to warning.

Comment (27Apoorva) [reply dwa_local_planner/src/dwa_planner_ros.cpp]: that means the function `resetLatching` will never be called?

Comment (ccpowers) [reply dwa_local_planner/src/dwa_planner_ros.cpp]: it's called in several other places in the code, but won't be hit here until we change wrapper_local_planner

File: base_local_planner/src/latched_stop_rotate_controller.cpp [hunk 2]
```diff
@@ -221,10 +241,17 @@ bool LatchedStopRotateController::computeVelocityCommandsStopRotate(geometry_msg
     ROS_INFO("Goal position reached, stopping and turning in place");
     xy_tolerance_latch_ = true;
   }
+
   //check to see if the goal orientation has been reached
   double goal_th = tf::getYaw(goal_pose.getRotation());
   double angle = base_local_planner::getGoalOrientationAngleDifference(global_pose, goal_th);
-  if (fabs(angle) <= limits.yaw_goal_tolerance) {
+  if ((fabs(angle) <= limits.yaw_goal_tolerance ) || (latch_yaw_goal_tolerance_ && yaw_tolerance_latch_)) {
```
Comment (dnikitop) [inline base_local_planner/src/latched_stop_rotate_controller.cpp:248]: latch_yaw_goal_tolerance_ && yaw_tolerance_latch_ is super confusing tbh

Comment (ccpowers) [reply base_local_planner/src/latched_stop_rotate_controller.cpp:248]: yeah but it's the same naming scheme we used for the xy tolerances as well

Comment (dnikitop) [reply base_local_planner/src/latched_stop_rotate_controller.cpp:248]: true, let's keep it

**Review by 27Apoorva — APPROVED** [review]

**Pull request merged**